.github/scripts/trymerge.py

```diff
@@ -1889,7 +1890,7 @@ def merge(
             ignore_flaky_failures = True
             try:
                 find_matching_merge_rule(
```

Comment:
Please double check if this works as expected as I have a gut feeling that having the below snipping running in `find_matching_merge_rule` would make the wrong behavior as this function is run inside a while loop:

```
checks = pr.get_checkrun_conclusions()
...
if skip_current_checks:
    _, failing = categorize_checks(checks, list(checks.keys()))
    ignore_current_checks = [x[0] for x in failing]
```

Basically, the while loop keeps running, so the list of failing checks could be growing.  Thus, `ignore_current_checks` could also grow.  This seems wrong as `merge -i` means ignoring only those failed checks at the time when the merge starts. cc @clee2000 for a second opinion